### DISCUSSION TITLE
When I use the command ikmvc to compile the. net core dynamic library, it prompts System IO. FileNotFoundException: netstandard, what should I do?

### DISCUSSION THREAD

**Discussion by MrJ6153** [ikvmnet/ikvm#520] [Q&A]
E:\env>ikvmc -target:library -out:HelloWorld.dll HelloWorld.class
IKVM.Tools.Importer (8.5.1+Branch.main.Sha.e48c511c8ce8c9e1c13ef0a9d0f2154195621be8)
Copyright ? 2023 Jeroen Frijters, Windward Studios, Jerome Haltom, Shad Storhaug

*** INTERNAL COMPILER ERROR ***

PLEASE FILE A BUG REPORT FOR IKVM.NET WHEN YOU SEE THIS MESSAGE

IKVM.Tools.Importer, Version=8.5.1.0, Culture=neutral, PublicKeyToken=13235d27fcbfff58
E:\env\8.5.1\IKVM-8.5.1-tools-netcoreapp3.1-win7-x64\
3.1.32 64-bit

System.IO.FileNotFoundException: netstandard
   at IKVM.Reflection.Universe.Load(String refname, Module requestingModule, Boolean throwOnError) in C:\work\ikvm\src\IKVM.Reflection\Universe.cs:line 547
   at IKVM.Tools.Importer.AssemblyResolver.Init(Universe universe, Boolean nostdlib, IList`1 references, IList`1 userLibPaths) in C:\work\ikvm\src\IKVM.Tools.Importer\AssemblyResolver.cs:line 96
   at IKVM.Tools.Importer.IkvmImporterInternal.Compile(String[] args) in C:\work\ikvm\src\IKVM.Tools.Importer\IkvmImporterInternal.cs:line 179
   at IKVM.Tools.Importer.IkvmImporterInternal.Execute(String[] args) in C:\work\ikvm\src\IKVM.Tools.Importer\IkvmImporterInternal.cs:line 112

**Comment by wasabii** [comment]
ikvmc requires -reference for all of the assemblies of the target framework you are intending to link against. Which would include basics like System.Private.CoreLib, System.Runtime, mscorlib, System.Collections, and the other ~60.

**Reply by MrJ6153** [reply]
Thank you for your answer! It's my responsibility, I didn't describe the problem clearly!
I have reviewed the content of doc in the latest code, but still do not know the purpose of bin and tools. I am currently using. net core 3.1, and I am not sure if there is any relevant user documentation or examples for explanation. Based on the blog posts I found online, I have taken the following steps
1. Downloaded IKVM-8.6.4-bin-netcoreapp3.1,IKVM-8.6.4-tools-netcoreapp3.1-win7-x64.zip
2. After decompression, remove the IKVM.Runtime.dll and IKVM.Java.dll from IKVM-8.6.4-bin-netcoreapp-3.1, placed in the IKVM-8.6.4-tools folder
3. Compiled a HelloWorld.jar file
4. Run command:    ikvmc.exe -target:library HelloWorld.jar
5. Tip: System.IO.FileNotFoundException: netstandard
Which step did I make an error in, or what do I need to install?

**Reply by wasabii** [reply]
Well, we no longer support .NET Core 3.1, for one.

I don't know your question. You asked why you got an exception. I told you.

**Reply by MrJ6153** [reply]
Okay, I tried a few more times later and found that although there are still errors in the console, the DLL file has been generated and can be referenced for use in the project. Thank you very much!
